include/deal.II/base/index_set.h

```diff
@@ -79,6 +89,188 @@ class IndexSet
   typedef signed int value_type;
 
   /**
+   * Dereferencing an IntervalIterator will return a reference to an object of this type.
+   */
+  class IntervalAccessor
+  {
+  public:
+    IntervalAccessor(const IndexSet &idxset, size_type range_idx);
+    size_type n_elements() const;
+    bool is_valid() const;
+    ElementIterator begin() const;
+    ElementIterator end() const;
+  private:
+    bool operator == (const IntervalAccessor &other) const;
+    bool operator < (const IntervalAccessor &other) const;
+    void advance ();
+    const IndexSet &index_set;
+    size_type range_idx;
+
+    friend class IntervalIterator;
+  };
+
+  /**
+   * Class that represents an iterator pointing to a contiguous interval [a,b[ as returned by IndexSet::begin_interval().
```

Comment:
use `$...$` to denote formulas. Maybe state somewhere that the `IndexSet` class stores index sets as a collection of half-open intervals, and that this class iterates over this interval.